src/Sarif/Visitors/ReformattingVisitor.cs

```diff
@@ -43,23 +45,129 @@ public override Run VisitRun(Run node)
+            // If we have a binary region, there is no work to do
+            if (node.Region == null ||
+                (node.Region.ByteOffset > 0 || node.Region.ByteLength > 0))
+            {
+                goto Exit;
+            }
+
+            // If we already have a snippet, let's assume it is correct
+            if (node.Region.Snippet != null)
+            {
+                goto Exit;
+            }
 
-            if (!Uri.TryCreate(key, UriKind.RelativeOrAbsolute, out uri))
+            // If the user hasn't specified code snippet insertion, there is no work to do
+            bool workToDo = false;
+            workToDo |= node.Region != null && _dataToInsert.Includes(OptionallyEmittedData.CodeSnippets);
+
+            if (!workToDo)
+            {
+                goto Exit;
+            }
+
+            // If we have an absolute URL specified as a file that 
+            // exist on disk, there is work to do
+            if (node.FileLocation.Uri.IsAbsoluteUri &&
+                node.FileLocation.Uri.LocalPath != null &&
+                File.Exists(node.FileLocation.Uri.LocalPath))
+            {
+                Region region = node.Region;
+                string fileText = GetFileText(node.FileLocation.Uri);
+                GetCompleteFirstLineAssociatedWithRegion(region, fileText);
+            }
+
+            Exit:
+            return base.VisitPhysicalLocation(node);
+        }
+
+        internal static void GetCompleteFirstLineAssociatedWithRegion(Region region, string fileText)
+        {
+            if (region.IsBinaryRegion)
+            {
+                return;
+            }
+
+            var lineIndex = new NewLineIndex(fileText);
+
+            region.Snippet = new FileContent();
+
+            int startLine;
+            LineInfo lineInfo;
+
+            if (region.StartLine == 0)
+            {
+                lineInfo = lineIndex.GetLineInfoForOffset(region.CharOffset);
+                startLine = lineInfo.LineNumber;
+            }
+            else
+            {
+                lineInfo = lineIndex.GetLineInfoForLine(region.StartLine);
+                startLine = region.StartLine;
+            }
+
+            int lineOffset = lineInfo.StartOffset;
+            int lineLength;
+
+            if (startLine == lineIndex.MaximumLineNumber)
+            {
+                // End of file. Compute length using file length
+                lineLength = fileText.Length - lineOffset;
+            }
+            else
+            {
+                // Not at end of file. Get the offset from the
+                // line after this one, then compute length from there.
+                lineInfo = lineIndex.GetLineInfoForLine(startLine + 1);
+                lineLength = lineInfo.StartOffset - lineOffset;
+            }
+
+            // Grab the line text, except for any trailing line break characters
+            region.Snippet.Text = fileText.Substring(lineOffset, lineLength).Trim();
```

Comment:
You're not just trimming the newline sequence; you're trimming trailing white space.

From an experience point of view, maybe that's ok (there's no point squiggling trailing spaces). But if that's what you intend, the comment should say so.

And in fact, you might have a style rule that says "No Trailing Whitespace", and then you'd want to squiggle it. So on further reflection, I think this is a bug.

Reply:
(I see you mentioned this issue in your initial comments. In any case, again, if we decide it's right to Trim, then then comment should reflect that.)

Reply:
Changed it just to trim the line break chars

---
In reply to: [202104581](https://github.com/Microsoft/sarif-sdk/pull/942#discussion_r202104581) [](ancestors = 202104581)